File: src/main/java/co/electric/snake/model/ModifiableSnake.java
package co.electric.snake.model;

import co.electric.snake.model.strategy.SnakeStrategy;

public class ModifiableSnake extends Snake {

    public ModifiableSnake(ModifiableArena arena, SnakeStrategy strategy, String name) {
        super(arena, strategy, name);
    }

    public void move() {
        final Coordinate nextCoordinate = decideNextCoordinate();
        moveTo(nextCoordinate);
    }

}

File: src/main/kotlin/co/electric/snake/model/ModifiableSnake.kt
package co.electric.snake.model

import co.electric.snake.model.strategy.SnakeStrategy

class ModifiableSnake(arena: ModifiableArena, strategy: SnakeStrategy, name: String) : Snake(arena, strategy, name) {

    fun move() {
        val nextCoordinate = decideNextCoordinate()
        moveTo(nextCoordinate)
    }

}
